README.md

```diff
@@ -214,15 +197,34 @@ undoable(reducer, { filter: excludeAction(SOME_ACTION) })
 // or you could do...
 
 undoable(reducer, { filter: distinctState() })
-```
 
-... they even support Arrays:
+// they even support Arrays:
 
-```js
 undoable(reducer, { filter: includeAction([SOME_ACTION, SOME_OTHER_ACTION]) })
 undoable(reducer, { filter: excludeAction([SOME_ACTION, SOME_OTHER_ACTION]) })
 ```
 
+If you want to create your own filter, pass in a function with the signature
+`(action, currentState, previousHistory)`. For example:
+
+```js
+undoable(reducer, {
+  filter: function filterActions(action, currentState, previousHistory) {
+    return action.type === SOME_ACTION; // only add to history if action is SOME_ACTION
+  }
+})
+
+// The entire `history` state is available to your filter, so you can make
+// decisions based on past or future states:
+
+undoable(reducer, {
+  filter: function filterState(action, currentState, previousHistory) {
+    let { past, present, future } = previousHistory;
+    return future.length === 0; // only add to history if future is empty
```

Comment:
I figure that it's a little more clear in the docs to show that it is a `previousHistory` and do the destructuring on the next line, because not everyone is totally comfortable with destructuring yet.